data/scripts.ts

```diff
@@ -787,39 +783,42 @@ let BattleScripts = {
 			}
 		}
 
-		for (let i = 0; i < targets.length; i++) {
+		for (let [i, t] of targets.entries()) {
 			if (damage[i] === 0) {
 				// special substitute flag
 				damage[i] = true;
 				targets[i] = null;
 			}
-			if (targets[i] && isSecondary && !moveData.self) {
+			if (t && isSecondary && !moveData.self) {
 				damage[i] = true;
 			}
-			if (!damage[i]) targets[i] = false;
+			if (!damage[i]) t = false;
 		}
 		// 1. call to this.getDamage
 		damage = this.getSpreadDamage(damage, targets, pokemon, move, moveData, isSecondary, isSelf);
 
-		for (let i = 0; i < targets.length; i++) {
-			if (damage[i] === false) targets[i] = false;
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		for (let [i, t] of targets.entries()) {
+			if (damage[i] === false) t = false;
```

Comment:
I'm just impressed that eslint caught a real bug and you just disabled it instead of trying to figure out what the bug was. ;)